pkg/controller/gitopsconfig/controller.go

```diff
@@ -417,3 +485,12 @@ func isOwner(owner, owned metav1.Object) bool {
 	}
 	return false
 }
+
+func getJobStatus(succeeded, failed int32) string {
+	if succeeded != 0 {
+		return "SUCCESS"
+	} else if failed != 0 {
+		return "FAILED"
+	}
+	return "INPROGRESS"
+}
```

Comment:
The general idea here seems to make sense. I would however suggest to improve this function in the following ways:

  1. If possible, to pass whole job status object as argument, and check the Succeeded and Failed fields inside this function, instead of passing 2 ints
  2. If `succeeded + failed > 1`, return error — we expect only 1 pod in the Job as of now, so if there are more of them, this means something is wrong